android/build.gradle

```
@@ -36,6 +40,10 @@ android {
     lintOptions {
         disable 'InvalidPackage'
     }
+    compileOptions {
+        targetCompatibility JavaVersion.VERSION_17
+        sourceCompatibility JavaVersion.VERSION_17
+    }
```

Needed for project using java.

```suggestion
    compileOptions {
        targetCompatibility JavaVersion.VERSION_17
        sourceCompatibility JavaVersion.VERSION_17
    }
    
    kotlinOptions {
        jvmTarget = "17"
    }
```

---

Is it strictly necessary to increase it to such a high Java Version? I wonder if this might lead to issues for other people as I don't think most (Flutter) developers will already be on Java 17. Would your issues also be fixed if you add the capability to Java Version 1.8? This it what `share_plus` does for example:
https://github.com/fluttercommunity/plus_plugins/blob/main/packages/share_plus/share_plus/android/build.gradle

---

AGP 8 requires Java 17 anyway https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes#jdk-17-agp, so it's strange that it can use 1.8 there :D 

Btw. locally I use home_widget with 

```
kotlinOptions {
    jvmTarget = "1.8"
}
```

(without specifying compileOptions source and target compatibility)

and with AGP 8 (`classpath 'com.android.tools.build:gradle:8.1.4'`)

and it works fine.

So I am not really sure what is correct to use there haha

---

Using java version 1.8 works for me.

---

I think I would prefer the 1.8 version then as this seems to be more in line how other packages handle that right now